### ISSUE TITLE
Reformat non-normative text in specification

### ISSUE THREAD

**Issue by pmai** [modelica/fmi-standard#459]
The current way that non-normative text is formatted in the specification (i.e. as both inline and seperate paragraphs in italics with bracketing) easily leads to the seperation between non-normative text and normative text getting lost. Additionally that way of formatting is not a good fit with the new formatting infrastructure.

I'd like to suggest to make all non-normative text separate paragraphs, with note admonitions for example, so that they are clearly seperated out (and formatting of multi-paragraph and advance content in non-normative parts easier to achieve).

E.g. using the note admonition (see http://asciidoc.org/userguide.html#X28) would look like this for a single paragraph:

    NOTE: This is non-normative text

or, using AttributeList:

    [NOTE]
    This is an example note.

And like this using adminition blocks for multi-paragraph text that can contain complex formatting: 

    [NOTE]
    .A NOTE admonition block
    =====================================================================
    Qui in magna commodo, est labitur dolorum an. Est ne magna primis
    adolescens.
    
    . Fusce euismod commodo velit.
    . Vivamus fringilla mi eu lacus.
      .. Fusce euismod commodo velit.
      .. Vivamus fringilla mi eu lacus.
    . Donec eget arcu bibendum
      nunc consequat lobortis.
    =====================================================================

**Comment by pmai** [comment]
That points to a problem of the document: Non-normative text should be droppable without loss to the defined semantics, otherwise it needs to be normative text.

So the solution would depend on each instance (turn it into normative text suitably altered, move it to a separate non-normative text block that repeats relevant parts of the table, or, if all else fails, have footnotes in the table referring to stuff in a non-normative block, ...).

In theory it should always be possible to remove all non-normative text from the document. The current approach leads to many of the conformance issues we are seeing.

Or in other words, the current normative parts are too weak, hence the reliance on too much non-normative text to explain the meaning of the normative parts...

At least in my experience... feel free to disagree.

And I’m not saying that this is not going to be a bit of work to get everything right, but I’d rather try to do something about this so that we have better chances at conformance meaning something.

PS: In many other standards only examples are considered non-normative, and more explanatory stuff is relegated to other documents for a reason ...

**Comment by clagms** [comment]
I second the change of the non normative text format.
For an example where the current approach looks bad, check out sections `2.1.11.1. Directional derivatives` and `2.1.11.2. Adjoint derivatives`. They consist almost exclusively of non-normative text, and I've corrected already some typos there because every line must be surrounded by `_`.
It is all too easy to forget this and end up with text that should be non normative, but isn't...

**Comment by andreas-junghanns** [comment]
> 
> 
> I second the change of the non normative text format.
> For an example where the current approach looks bad, check out sections `2.1.11.1. Directional derivatives` and `2.1.11.2. Adjoint derivatives`. They consist almost exclusively of non-normative text, and I've corrected already some typos there because every line must be surrounded by `_`.
> It is all too easy to forget this and end up with text that should be non normative, but isn't...

The last point is easy: use a suitable editor that helps you with proper syntax highlighting. Or, as in our case, we should all use the same broken debugger (Atom) that will highlight in the same broken way.

Just changing the format might help readability, but will not address Pierre´s concerns: some non-normative notes contain restrictions that should be normative.

I would primarily worry about the later point, the former (format) is nice to have.

**Comment by clagms** [comment]
> some non-normative notes contain restrictions that should be normative.

I understand this, and already found some.

However, I think we should still consider adopting a multi-paragraph environment to represent multiple paragraphs (with tables, figs, etc.) of non normative text.
Another example: search for the text "some non-normative notes contain restrictions that should be normative." an you will note that there are multiple paragraphs and non normative text is in italic, but is not correctly contained between two `[]`.
I checked this with VS Code (that has a adoc plugin) but it was not obvious that this mistake is there.
Also, relying on a suitable editor to detect this things is bound to fail, as people like to use different editors. If we have a proper environment, chances are that the adoc compiler will show a syntax error when these mistakes happen.

Regarding the original concern:
> ... easily leads to the seperation between non-normative text and normative text getting lost.

I suggest that when we move remarks out of the text (e.g., from tables and all that)., we can refer to them by a link. Kind of like a footnote in a latex document.
To give an illustration, suppose we have a table in which some columns have non normative text.
We should move all non normative text to an environment that comes after the table, and, in this environment, we can have a numbered list of all remarks. Then the text that needs to refer to the remarks just uses the right label.

If it's too hard to have labels and refs for every remark after the table, we can just keep a more generic referes (e.g., `(see remarks below this table)`) and make it clear in each remark what column it refers to.

Does this make sense?

**Comment by andreas-junghanns** [comment]
All of the above makes sense, Cláudio - but let´s consider what else we need to do and how important this is compared to the other tasks. I think, the cleanup you started is more important for now and once this is through, we will need to check if such a HUGE undertaking is worth the risk of introducing other issues (which invariably happens with such big changes).

Again: I am not saying this would make no sense - on the contrary. But we are still fighting fundamental issues: Tables that are not filled in, text passages that are missing, descriptions that need to be written, some 50+ TODOs in the text point to these priority 1 issues. Can we maybe do those first?

**Comment by chrbertsch** [comment]
Ticket-viewing-meeting

Andreas: we have several inconsistencies.
TorstenB: Should this be part of our write-fest?
Andreas: Could come later, in parallel with the language checking
TorstenB: But we should mark  things that are normative
Christian: We could already think about a better technical solution for formatting of non-normative texst in ADOC.
@t-sommer : could you please look into this or close this ticket?

**Comment by t-sommer** [comment]
I suggest to close this issue.
